python/tests/unit/contents/test_chat_history.py

```diff
@@ -281,6 +274,62 @@ async def test_template(chat_history: ChatHistory):
     assert chat_history_2.messages[2].role == ChatRole.USER
 
 
+@pytest.mark.asyncio
+async def test_chat_history_with_message_type():
+    chat_history = ChatHistory(message_type="OpenAIChatMessageContent")
```

Comment:
As a dev, I am going to now need to supply the message type when I create a chat history?

Reply:
You can, but you don't have to, the ChatHistory does not care what kind of message is in there, as long as it is a CMCBase thing, during serialization (both to json and xml) the types of each message are passed along and when rebuilding the type of the whole history is used, if not specified, defaults to regular CMC. But if you want to have OpenAICMC's as every message (because you want to add stuff during hooks or something, you have the option!).